crates/holochain/src/core/sys_validate.rs

```diff
@@ -202,20 +204,18 @@ pub async fn check_app_entry_type(
     let zome_index = u8::from(entry_type.zome_id()) as usize;
     // We want to be careful about holding locks open to the conductor api
     // so calls are made in blocks
-    let dna_file = { conductor_api.get_this_dna().await };
-    let dna_file =
-        dna_file.ok_or_else(|| SysValidationError::DnaMissing(conductor_api.cell_id().clone()))?;
+    let dna_file = conductor_api.get_this_dna().await.map_err(Box::new)?;
```

Comment:
This isn't really related to your PR but I wonder if it's bad that you could send someone an op with a nonsense dna and panic there validation thread :thinking:

Reply:
Guess we need to have a recoverable vs fatal error audit at some point